src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionMiddleware.cs

```diff
@@ -127,16 +143,34 @@ private void CheckForHttpsPorts()
                     // If we find multiple different https ports specified, throw
                     if (httpsPort.HasValue && httpsPort != bindingAddress.Port)
                     {
-                        throw new ArgumentException("Cannot determine the https port from IServerAddressesFeature, multiple values were found. " +
+                        _logger.LogWarning("Cannot determine the https port from IServerAddressesFeature, multiple values were found. " +
                             "Please set the desired port explicitly on HttpsRedirectionOptions.HttpsPort.");
+
+                        _portEvaluated = true;
+                        port = -1;
+                        return false;
                     }
                     else
                     {
                         httpsPort = bindingAddress.Port;
                     }
                 }
             }
-            _httpsPort = httpsPort ?? 443;
+
+            if (httpsPort.HasValue)
+            {
+                _logger.LogDebug($"Https port '{httpsPort}' discovered from server endpoints.");
```

Comment:
Stay consistent on capitalization of HTTPS.

Reply:
Where is it "HTTPS"? It's "https" everywhere except at the start of a sentence.